apollo-router/src/configuration/mod.rs

```diff
@@ -690,7 +690,8 @@ pub(crate) struct Supergraph {
     pub(crate) reuse_query_fragments: Option<bool>,
 
     /// Enable QP generation of fragments for subgraph requests
-    /// Default: false
+    /// Default: true
+    #[serde(default = "default_generate_query_fragments")]
```

Comment:
this fails our own `default` tests:

<img width="1009" alt="Screenshot 2024-09-17 at 14 59 07" src="https://github.com/user-attachments/assets/11ae4016-ce60-4dee-9b37-ddfb6a0c7d57">

The only way for this config to have its own `Default` implementation is to be defined as an enum/struct, which feels wayyy too much for something like this. Any other ideas on setting up a default in a single place for this config?

Reply:
Remove `#[serde(default = "default_generate_query_fragments")]` and replace two instances of `generate_query_fragments.unwrap_or_default()` in this file with `generate_query_fragments.unwrap_or_else(default_generate_query_fragments)`

Since the struct already has `#[serde(default)]` serde will rely on the existing `impl Default for Supergraph`

Reply:
i felt that using `unwrap_or_else` hid away the default definition. but i guess since we've got a doc comment with `Default: true` it's maybe a bit more clear.